src/osd/OSD.cc

```diff
@@ -1718,10 +1726,27 @@ void OSDService::send_ready_to_merge()
 
 void OSDService::_send_ready_to_merge()
 {
+  for (auto src : not_ready_to_merge_source) {
+    if (sent_ready_to_merge_source.count(src) == 0) {
+      monc->send_mon_message(new MOSDPGReadyToMerge(
+			       src,
+			       0,
+			       false,
+			       osdmap->get_epoch()));
```

Comment:
Need to move these(PGS) to `sent_ready_to_merge_source `too so we can resend if the connection is reset

Reply:
fixed!